providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py

```diff
@@ -394,7 +394,7 @@ def run_next(self, next_job: KubernetesJobType) -> None:
 
             if isinstance(command[0], ExecuteTask):
                 workload = command[0]
-                ser_input = workload.model_dump_json()
+                ser_input = workload.model_dump_json(exclude={"ti": {"executor_config"}})
```

Comment:
Why do we need to exclude this here?

Reply:
`executor_config` is of `k8s.V1Pod` type. We need to exclude it from the serialisation entirely because pydantic doesnt know how to serialise `k8s.V1Pod` types.  A few lines later, it is passed onto the `construct_pod`

Reply:
Please add a comment saying why we are doing this then. I agree it makes sense that we don't need to pass it in to the pod though.

(FWIW ariflow.serialzation.serde's `serialize` function would handle this. But not passing it is the right call here)

Reply:
Added in https://github.com/apache/airflow/pull/46926/commits/4cc4fa6004d5af24eec591e73e78ca0654487bda